### PR TITLE
Use dual-stack VIPs for dual-stack vSphere clusters

### PR DESCRIPTION
**Pull request by mkowalski**
Contributes-to: OCPBUGS-12739

### CODE REVIEW

**Comment by mkowalski** [comment]
/hold
WIP

**Comment by mkowalski** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn-dualstack

**Comment by mkowalski** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn-dualstack

**Comment by mkowalski** [comment]
The following install-config has been generated for this run

```
apiVersion: v1
metadata:
  name: ci-op-3cl534p4-e6d12
  {"auths":{"clo[...]hat.com"}}}
sshKey: |
  ssh-rsa AAAA[...]w==
baseDomain: vmc-ci.devcluster.openshift.com
platform:
  vsphere:
    [...]
    apiVIPs:
      - 192.168.109.2
      - fd65:a1a8:60ad::2
    ingressVIPs:
      - 192.168.109.3
      - fd65:a1a8:60ad::3
networking:
  networkType: OVNKubernetes
  machineNetwork:
  - cidr: 192.168.0.0/16
  - cidr: fd65:a1a8:60ad::/112
  clusterNetwork:
  - cidr: 10.128.0.0/14
    hostPrefix: 23
  - cidr: fd65:10:128::/56
    hostPrefix: 64
  serviceNetwork:
  - 172.30.0.0/16
  - fd65:172:16::/112
  - ```

**Comment by mkowalski** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-master-e2e-vsphere-ovn-dualstack
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.14-e2e-vsphere-ovn-dualstack

**Comment by mkowalski** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn-dualstack

File: ci-operator/step-registry/ovn/conf/vsphere-dualstack/ovn-conf-vsphere-dualstack-commands.sh [hunk 1]
```diff
@@ -3,11 +3,25 @@ set -o errexit
...
+
 echo "removing networking config if already exists"
 /tmp/yq e --inplace 'del(.networking)' ${SHARED_DIR}/install-config.yaml
+/tmp/yq e --inplace 'del(.platform.vsphere.apiVIP)' ${SHARED_DIR}/install-config.yaml
+/tmp/yq e --inplace 'del(.platform.vsphere.ingressVIP)' ${SHARED_DIR}/install-config.yaml
 
 echo "applying dual-stack networking config"
 
+/tmp/yq e --inplace '.platform.vsphere.apiVIPs += [strenv(API_VIP)]' ${SHARED_DIR}/install-config.yaml
+/tmp/yq e --inplace '.platform.vsphere.apiVIPs += ["fd65:a1a8:60ad::2"]' ${SHARED_DIR}/install-config.yaml
```
Comment (rbbratta) [inline ci-operator/step-registry/ovn/conf/vsphere-dualstack/ovn-conf-vsphere-dualstack-commands.sh:21]: We could maybe combine these?  
```
.platform.vsphere.apiVIPs += [strenv(API_VIP), "fd65:a1a8:60ad::2"]
```

I'll have to check with `yq`

**Comment by mkowalski** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn-dualstack

**Comment by mkowalski** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-master-e2e-vsphere-ovn-dualstack

Master nodes fail to obtain the ignition from `api-vip:some-port/config/master`. Investigating manually now inside bootstrap VM

**Comment by mkowalski** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-master-e2e-vsphere-ovn-dualstack
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.14-e2e-vsphere-ovn-dualstack

That was a stupid human error. I was configuring VIPs from outside of Machine Network. The nodes live in `fd65:a1a8:60ad:271c:f2bb:2f5b:ad41:c8d1/64` so `fd65:a1a8:60ad::2` was obviously from outside of this error. Because of this, keepalived was failing to start on the bootstrap VM.

I have now fixed it, machines live in `fd65:a1a8:60ad:271c::/64` and all the addresses should reflect it.

**Comment by rbbratta** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-ovn-dualstack-p2-f7

**Comment by cybertron** [comment]
/lgtm

I can't see the qe job results so I'm not sure what happened there, but this seems like a sane change.

**Comment by rbbratta** [comment]
/lgtm 

please ignore the failed qe tests-private.

**Pull request merged**